evap/staff/importers.py

```diff
@@ -689,21 +731,40 @@ def process_participants(self, evaluation, test_run, user_list):
 
         if already_related:
             msg = format_html(
-                _("The following {} users are already participants in evaluation {}:"),
-                len(already_related),
-                evaluation.full_name,
+                ngettext(
+                    "The following user is already participant in evaluation {name}:",
+                    "The following {users} are already participants in evaluation {name}:",
+                    len(already_related),
+                ).format(users=len(already_related), name=evaluation.full_name)
             )
             msg += create_user_list_html_string_for_message(already_related)
             self.warnings[ImporterWarning.GENERAL].append(msg)
 
-        if not test_run:
-            evaluation.participants.add(*users_to_add)
-            msg = format_html(_("{} participants added to the evaluation {}:"), len(users_to_add), evaluation.full_name)
+        if len(users_to_add) == 0:
+            if not test_run:
+                msg = format_html(_("No participants added to the evaluation {}"), evaluation.full_name)
+            else:
+                msg = format_html(_("No participants would be added to the evaluation {}"), evaluation.full_name)
+
         else:
-            msg = format_html(
-                _("{} participants would be added to the evaluation {}:"), len(users_to_add), evaluation.full_name
-            )
-        msg += create_user_list_html_string_for_message(users_to_add)
+            if not test_run:
+                evaluation.participants.add(*users_to_add)
+                msg = format_html(
+                    ngettext(
+                        "1 participant added to the evaluation {name}:",
+                        "{users} participants added to the evaluation {name}:",
+                        len(users_to_add),
+                    ).format(name=evaluation.full_name, users=len(users_to_add))
+                )
+            else:
+                msg = format_html(
+                    ngettext(
+                        "1 participant would be added to the evaluation {name}:",
+                        "{users} participants would be added to the evaluation {name}:",
+                        len(users_to_add),
+                    ).format(name=evaluation.full_name, users=len(users_to_add))
+                )
```

Comment:
There are multiple places with this much duplication, just for the word "would". Is there any chance that one could define
```python
if test_run:
    would_word = "would be "
else:
    would_word = ""
# ...
x = f"1 participant {would_word}added to the evaluation"
```
or similar? Or is that too much trouble for translation @janno42 ?

Reply:
fyi he said its complicated

Reply:
right, we can't assume there is just one word because the german translation can have a splitted verb ("wird hinzugefügt"/"würde hinzugefügt werden").
therefore translations always need to be complete phrases.